homeassistant/components/freebox/alarm_control_panel.py

```diff
@@ -0,0 +1,170 @@
+"""Support for Freebox alarms."""
+import logging
+from typing import Any
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_ARMING,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+    STATE_IDLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, FreeboxHomeCategory
+from .home_base import FreeboxHomeEntity
+from .router import FreeboxRouter
+
+FREEBOX_TO_STATUS = {
+    "alarm1_arming": STATE_ALARM_ARMING,
+    "alarm2_arming": STATE_ALARM_ARMING,
+    "alarm1_armed": STATE_ALARM_ARMED_AWAY,
+    "alarm2_armed": STATE_ALARM_ARMED_NIGHT,
+    "alarm1_alert_timer": STATE_ALARM_TRIGGERED,
+    "alarm2_alert_timer": STATE_ALARM_TRIGGERED,
+    "alert": STATE_ALARM_TRIGGERED,
+}
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up alarm panel."""
+    router: FreeboxRouter = hass.data[DOMAIN][entry.unique_id]
+
+    alarm_entities: list[AlarmControlPanelEntity] = []
+
+    for node in router.home_devices.values():
+        if node["category"] == FreeboxHomeCategory.ALARM:
+            alarm_entities.append(FreeboxAlarm(hass, router, node))
+
+    if alarm_entities:
+        async_add_entities(alarm_entities, True)
+
+
+class FreeboxAlarm(FreeboxHomeEntity, AlarmControlPanelEntity):
+    """Representation of a Freebox alarm."""
+
+    def __init__(
+        self, hass: HomeAssistant, router: FreeboxRouter, node: dict[str, Any]
+    ) -> None:
+        """Initialize an alarm."""
+        super().__init__(hass, router, node)
+
+        self._attr_extra_state_attributes = {}
+        self._state: str | Any
+        # Trigger
+        self._command_trigger = self.get_command_id(
+            node["type"]["endpoints"], "slot", "trigger"
+        )
+        # Alarme principale
+        self._command_alarm1 = self.get_command_id(
+            node["type"]["endpoints"], "slot", "alarm1"
+        )
+        # Alarme secondaire
+        self._command_alarm2 = self.get_command_id(
+            node["type"]["endpoints"], "slot", "alarm2"
+        )
+        # Passer le délai
+        self._command_skip = self.get_command_id(
+            node["type"]["endpoints"], "slot", "skip"
+        )
+        # Désactiver l'alarme
+        self._command_off = self.get_command_id(
+            node["type"]["endpoints"], "slot", "off"
+        )
+        # État
+        self._command_state = self.get_command_id(
+            node["type"]["endpoints"], "signal", "state"
+        )
+
+        self.set_state(STATE_IDLE)
```

Comment:
This state will never work as not handled in `set_state` + the alarm is updated at its creation with the `True` parameter of `async_add_entities`